components/brave_today/browser/publishers_parsing.cc

```diff
@@ -39,6 +40,22 @@ bool ParseCombinedPublisherList(const std::string& json,
     publisher->publisher_name = *publisher_raw.FindStringKey("publisher_name");
     publisher->category_name = *publisher_raw.FindStringKey("category");
     publisher->is_enabled = publisher_raw.FindBoolKey("enabled").value_or(true);
+    GURL feed_source(*publisher_raw.FindStringKey("feed_url"));
+    if (feed_source.is_valid()) {
+      publisher->feed_source = feed_source;
+    }
+
+    std::string site_url_raw = *publisher_raw.FindStringKey("site_url");
```

Comment:
To work properly, maybe data from our service also should have `site_url` property for each combined source?

Reply:
and we don't need to handle this also when direct publisher parsing?

Reply:
I think changes in this files also should be guarded.

Reply:
So currently, all data from our backend service does have a site_url (see https://brave-today-cdn.brave.com/sources.json), we just weren't parsing it previously because we weren't using it.

Direct publisher parsing is more complicated - we don't always have a `site_url` to work with (as you might pass us just the url for the feed, rather than a page on the site). It might be worth adding it in as part of a future change, but I don't think it's too urgent yet.

Guarded, as in "Only parse these fields if the feature flag is set"? Happy to add logic for that if you think it's worth it, but as these fields are already present I don't know if it's necessary.

Reply:
Ok,  I was curious backend data includes `site_url` props or not.  As it already has, I think guard is not mandatory. let's go as-is.